### PR TITLE
[5.1] [Name lookup] Deduplicate nominal declarations found via resolveTypeDeclsToNominal

### PR DESCRIPTION
**Pull request by DougGregor**
Perform basic deduplication of the nominal type declarations found via
resolveTypeDeclsToNominal(). Fixes rdar://problem/53164203

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by slavapestov** [comment]
Does this also address rdar://51413553 ?

File: test/NameBinding/Inputs/property_wrappers_A.swift [hunk 1]
```diff
@@ -0,0 +1,8 @@
+@propertyWrapper
+public struct Wrapper<Value> {
+	public var wrappedValue: Value
```
Comment (slavapestov) [inline test/NameBinding/Inputs/property_wrappers_A.swift]: Tabs again? :)

Comment (DougGregor) [reply test/NameBinding/Inputs/property_wrappers_A.swift]: Didn't cherry-pick over https://github.com/apple/swift/pull/26178 yet

**Pull request merged**